### ISSUE TITLE
Some promises on Android never resolve when using React Native 0.65.x or under.

### ISSUE THREAD

**Issue by talhamirza1** [stripe/stripe-react-native#1095]
Hey, I am using stripe to handle payments from react native applications
I am using the official documentation provided by stripe. Everything works fine on iOS and the payment is going through the stripe dashboard I can see the payment successful.

Steps to Reproduce:

1. create payment intent 
2. get client secret
3. init payment sheet 
4. and finally presentPayment sheet

Present payment Sheet does not resolve.
if there is an error it does work and shows the alert. but in case of successful payment it never resolves.
 
Docs Ref:

[https://stripe.com/docs/payments/accept-a-payment?platform=react-native](https://stripe.com/docs/payments/accept-a-payment?platform=react-native)

```
 const openPaymentSheet = async (clientSecretData) => {
    try {
      const { error: err } = await initPaymentSheet({
        paymentIntentClientSecret: clientSecretData,
      });
      if (err) {
        Alert.alert(`Error Code: ${err.code}`, err.message);
      } else {
        setAble(false);
        const { error } = await presentPaymentSheet({
          clientSecret: clientSecretData,
        });
        if (error) {
          Alert.alert(`Error Code: ${error.code}`, error.message);
        } else {
         // Do something here
console.log("Payment Successfull");Ï
        }
      }
    } catch (error) {
      Alert.alert(`Error Code: ${error.code}`, error.message);
    }
  };
```
**Expected behavior**
in case of Successful payment the else condition should work and print the message.

**Actual behavior**
await presentPaymentSheet
Never Resolves for android but for ios it does resolve and shows the message.

 - OS: Mac OS and Windows
 - android SDK 31
 - stripe Version 17.0

**Comment by charliecruzan-stripe** [comment]
Hi! This was fixed in [v0.18.1](https://github.com/stripe/stripe-react-native/commit/9687588e18c4b3e059a89aa687f4099681a570e8) :)

**Comment by talhamirza1** [comment]
> Hi! This was fixed in [v0.18.1](https://github.com/stripe/stripe-react-native/commit/9687588e18c4b3e059a89aa687f4099681a570e8) :) 

Oh awesome I just saw the changelog 
Thanks for that 🥳

**Comment by daveadapptor** [comment]
I'm seeing this on v0.19.0 running on an Android emulator under test mode.  I haven't tried on an actual device.  Seems that it does come back EVENTUALLY but its horribly slow, taking 10-15 seconds to come back.
